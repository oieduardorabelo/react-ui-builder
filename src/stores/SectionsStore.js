export default {
  get(section) {
    if (section === 'section1') {
      return {
        type: '1',
        data: {
          title: '',
          subTitle: '',
          linkImage: '',
        },
        edit: true,
      }
    }
    if (section === 'section2') {
      return {
        type: '2',
        data: {
          title: '',
          subTitle: '',
          backgroundColor: '',
          buttonLink: '',
          buttonTarget: '_self',
          buttonLabel: '',
        },
        edit: true,
      }
    }
    if (section === 'section3') {
      return {
        type: '3',
        data: {
          createdAt: '',
          linkImage1: '',
          linkImage2: '',
          linkImage3: '',
        },
        edit: true,
      }
    }

    throw new Error(`This section doesn't exist: ${section}`)
  },
}
