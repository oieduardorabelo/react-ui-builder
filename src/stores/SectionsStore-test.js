import SectionsStore from './SectionsStore'

describe('SectionsStore', () => {
  it('retursn Section 1 correctly', () => {
    const actual = SectionsStore.get('section1')
    expect(actual.type).toBe('1')
  })

  it('retursn Section 2 correctly', () => {
    const actual = SectionsStore.get('section2')
    expect(actual.type).toBe('2')
  })

  it('retursn Section 3 correctly', () => {
    const actual = SectionsStore.get('section3')
    expect(actual.type).toBe('3')
  })

  it('retursn correct error', () => {
    try {
      SectionsStore.get('section4')
    } catch (err) {
      expect(err).toBeDefined()
    }
  })
})
