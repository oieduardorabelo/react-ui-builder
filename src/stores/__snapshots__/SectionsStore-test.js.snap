// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SectionsStore retursn Section 1 correctly 1`] = `
Object {
  "data": Object {
    "linkImage": "",
    "subTitle": "",
    "title": "",
  },
  "edit": true,
  "type": "1",
}
`;

exports[`SectionsStore retursn Section 2 correctly 1`] = `
Object {
  "data": Object {
    "backgroundColor": "",
    "buttonLabel": "",
    "buttonLink": "",
    "buttonTarget": "_self",
    "subTitle": "",
    "title": "",
  },
  "edit": true,
  "type": "2",
}
`;

exports[`SectionsStore retursn Section 3 correctly 1`] = `
Object {
  "data": Object {
    "createdAt": "",
    "linkImage1": "",
    "linkImage2": "",
    "linkImage3": "",
  },
  "edit": true,
  "type": "3",
}
`;

exports[`SectionsStore retursn correct error 1`] = `[Error: This section doesn't exist: section4]`;
